js/compiler.ts

```diff
@@ -407,43 +416,60 @@ export class DenoCompiler
     if (!recompile && moduleMetaData.outputCode) {
       return moduleMetaData.outputCode;
     }
-    const { fileName, sourceCode, moduleId } = moduleMetaData;
+    const { fileName, sourceCode, mediaType, moduleId } = moduleMetaData;
     console.warn("Compiling", moduleId);
     const service = this._service;
-    const output = service.getEmitOutput(fileName);
-
-    // Get the relevant diagnostics - this is 3x faster than
-    // `getPreEmitDiagnostics`.
-    const diagnostics = [
-      ...service.getCompilerOptionsDiagnostics(),
-      ...service.getSyntacticDiagnostics(fileName),
-      ...service.getSemanticDiagnostics(fileName)
-    ];
-    if (diagnostics.length > 0) {
-      const errMsg = this._ts.formatDiagnosticsWithColorAndContext(
-        diagnostics,
-        this
-      );
-      console.log(errMsg);
-      // All TypeScript errors are terminal for deno
-      this._os.exit(1);
-    }
+    // Instead of using TypeScript to transpile JSON modules, we will just do
+    // it directly.
+    if (mediaType === MediaType.Json) {
+      // tslint:disable-next-line:max-line-length
+      moduleMetaData.outputCode = `define([], function() { return JSON.parse(\`${sourceCode}\`); });\n//# sourceURL=${fileName}`;
```

Comment:
Add a separate function for this.
```
export function jsonAmdTemplate(jsonStr: string): string
```
Add a test in compiler_test that does something like this
```
import { jsonAmdTemplate } from "./compiler";

test(function testJsonAmdTemplate() {
  const code = jsonAmdTemplate('{ "hello": "world", "foo", "bar" }');
  const result = eval(code);
  /// test result.
});

```